eng/scripts/typespec/Generate-Code.ps1

```diff
@@ -36,6 +36,8 @@ function Refresh-Build {
 $testProjectsLocalDir = Join-Path $packageRoot 'generator' 'TestProjects' 'Local'
 $basicTypespecTestProject = Join-Path $testProjectsLocalDir "Basic-TypeSpec"
 
+Push-Location $packageRoot
```

Comment:
This is to fix error during local test generation

Reply:
For local test generation, we need to run this script, but the script location is `eng/scripts/typespec/`
We need to set the working directory to `eng/packages/http-client-csharp`, which is `$packageRoot`

Reply:
cc @hallipr

Reply:
But where in the script are we running `npm run build`? I'm trying to understand why it would work in CI but not locally.

Reply:
@JoshLove-msft 
Right, let me explain more details about this.
>But where in the script are we running npm run build?

[npx tsp compile](https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Generate-Code.ps1#L15) and [npm run build](https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Generate-Code.ps1#L29) are not running in the expected working directory while running the script locally from "/eng/scripts/typespec/Generate-Code.ps1".

>I'm trying to understand why it would work in CI but not locally.

The reason it works in CI is due to the way we invoke this script in another script
https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Check-CodeGeneration.ps1#L6

As we can see, the working directory has been set to `$PSScriptRoot`, that's what I am trying to do in the script directly.
And I just realized with this update in place, we can just remove the directory from the place we invoke this script.

Reply:
@JoshLove-msft 
Right, let me explain more details about this.
>But where in the script are we running npm run build?

[npx tsp compile](https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Generate-Code.ps1#L15) and [npm run build](https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Generate-Code.ps1#L29) are not running in the expected working directory while running the script locally from "/eng/scripts/typespec/Generate-Code.ps1".

>I'm trying to understand why it would work in CI but not locally.

The reason it works in CI is due to the way we invoke this script in another script
https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Test-Emitter.ps1#L15-L18
And then the script is running in the expected working directory as below:
https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Test-Emitter.ps1#L63
https://github.com/Azure/azure-sdk-for-net/blob/313748c7fa6ef1a62639895d568fc36dd13ac719/eng/scripts/typespec/Check-CodeGeneration.ps1#L6

As we can see, the working directory has been set to `$packageRoott`, that's exactly what I am trying to do in the script directly.